src/sentry/ingest/partitioning.py

```diff
@@ -0,0 +1,43 @@
+"""
+The number of logical partitions used to distinguish between where records
+should be stored. These do not require individual physical slices but allow
+for repartitioning with less code changes per physical change.
+"""
+from django.conf import settings
+
+
+def map_org_id_to_logical_partition(org_id: int) -> int:
+    """
+    Maps an org_id to a logical partition. Since SENTRY_SLICING_LOGICAL_PARTITION_COUNT is
+    fixed, an org id will always be mapped to the same logical partition.
+    """
+    return org_id % settings.SENTRY_SLICING_LOGICAL_PARTITION_COUNT
+
+
+def map_logical_partition_to_slice(dataset_name: str, logical_partition: int) -> int:
+    """
+    Maps a logical partition to a slice.
+    """
+
+    assert is_storage_partitioned(
+        dataset_name
+    ), f"cannot retrieve slice of non-partitioned dataset {dataset_name}"
+    assert (
+        dataset_name in settings.SENTRY_SLICING_CONFIG
+    ), f"logical partition mapping missing for {dataset_name}"
+
+    for ((logical_part_lo_incl, logical_part_hi_excl), slice_id) in settings.SENTRY_SLICING_CONFIG[
+        dataset_name
+    ].items():
+        if logical_partition >= logical_part_lo_incl and logical_partition < logical_part_hi_excl:
+            return slice_id
+
+    assert False, f"no mapping for dataset {dataset_name} for logical_partition {logical_partition}"
+
+
+def is_storage_partitioned(dataset_name: str) -> bool:
```

Comment:
Based on my previous comment, now I definitely feel storage in not the right name. You could be partitioning any resource in Sentry like consumers, redis etc.

Reply:
Yeah, I think unfortunately there isn't a good cross-system term for this. What do you think of

```python
type Sliceable = Literal["generic_metrics_sets", "generic_metrics_distributions"]

def map_org_id_to_logical_partition(org_id: int) -> int: # no change
  ...

def map_logical_partition_to_slice(sliceable: Sliceable, logical_partition: int) -> int:
  ...

def is_storage_partitioned(sliceable: Sliceable) -> bool:
  ...
```

Reply:
maybe the last one could just be named `is_sliced`

Reply:
> Yeah, I think unfortunately there isn't a good cross-system term for this. What do you think of
> 
> ```python
> Sliceable = Literal["generic_metrics_sets", "generic_metrics_distributions"]
> 
> def map_org_id_to_logical_partition(org_id: int) -> int: # no change
>   ...
> 
> def map_logical_partition_to_slice(sliceable: Sliceable, logical_partition: int) -> int:
>   ...
> 
> def is_sliceable_partitioned(sliceable: Sliceable) -> bool:
>   ...
> ```
Yes that is better